### ISSUE TITLE
Specification and prototype of Success Case collectors

### ISSUE THREAD

**Issue by vecna** [globaleaks/GlobaLeaks#1178]
From #232:

SuccessCase: when a node or a receiver publish a news based on a submission. Is important that inside the globaleaks node, the admin or the receiver, report that success case. link + when + description. This can be used to display the success cases of the nodes. will be exported publicly by an API, in the beginning, not used, until leakdirectory will not fetch that.

**Comment by vecna** [comment]
please feedback on the specification above @evilavli3 @fpietrosanti @hellais

**Comment by evilaliv3** [comment]
i've never requested a specification of the micro feature but at least that the macro topics of the project are properly defined and after that there is a plan and a vision of what we want to do start addressing the.

this is due to the fact that on each deliverable we have an amount of time allocated (e.g. D7.3 see only 40 men days allocates) so that it's important to have a roadmp and a clear idea of the things that we want to see addressed at the end of this limited time.

but ok if you want start from D7 that we have currenly not defined this is the things that i would like to see defined before start.

https://docs.google.com/a/apps.globaleaks.org/document/d/1eCDxHfNrxe5Wwvy9Le-qd6_WLEThrWqANfPhzhAy-dM/edit

1) what is the aim of D7.3 and what should we address?
2) what are the main topics/features we envision to implement in D7.3 and what should be the roadmap of implementation.

after that we have a clear idea of 1) and 2) i'm ok to procced reading the spec that before that is simply proposing a puzzle piece of a picture i do not know.

**Comment by fpietrosanti** [comment]
That feature would be ineffective until some major big developments will happen, so let's postpone or it will just become a dead piece of code and UI (because not user will use it, so no effective testing will ever happen).

Please just postpone it.

Additionally, understanding the meaning / usefulness of "let's collect success case of an initiative" because it's something that "adopters" should use, please inquiry them IF they will ever use the feature that way and where do they expect to load that kind of information (on GlobaLeaks, that handle very sensitive data, or on LeakDirectory that handle public data).

I don't think that a receiver using GlobaLeaks for sensitive data, would ever use it to add public data (that as adopter, is going to add to my own public site).

**Comment by fpietrosanti** [comment]
Regarding the last point, i just made a feature discussion with Lorenzo from IRPI that gave very useful info:
- He would like to handle this data from receiver panel (not from admin)
- He would like to have the "public profiles" on LeakDirectory to be two separate, one for IRPILeaks and one for ExpoLeaks, that in their use-case are separate context of GlobaLeaks (like for OCCRPLeaks)

I didn't asked for "which information" do they expect to insert, but a sketch/mockup interface of the Leakdirectory side and the GlobaLeaks side will be required in order to identify/define which kind of data is required, getting a feedback from adopters.

The very same inquiry would be useful to be done/collected from other adopters, to collect their hypotetical practical use-case scenario.

**Comment by vecna** [comment]
@fpietrosanti @evilaliv3 answers inline:

>> That feature would be ineffective until some major big developments will happen, so let's postpone or it will just become a dead piece of code and UI 

I don't think so, based on the list of HS we can (interally) crawl that data and start to display, for presentation (like, at IJF) the effects of GlobaLeaks after three years from the release.

>> Additionally, understanding the meaning / usefulness of "let's collect success case of an initiative" because it's something that "adopters" should use, please inquiry them IF they will ever use the feature that way and where do they expect to load that kind of information

That's good indeed. but, I guess, this is part of **our** vision on how node has to provide feedback inside the network of node. Adopters can give a shit about, but we know that we need. 

>> I personally don't think that a receiver using GlobaLeaks for sensitive data, would use it to add such public bit of information

we are speaking of "report a link" I don't see such problem. we are not speaking of making sensitive stuff available. but I agree that other models can exists. For example: the receiver logs in LeakDirectory to update their own page. (I discourage this option because lead to a centralize model since the first day)


>> Regarding the last point, i just made a feature discussion with Lorenzo from IRPI that gave very useful info:
>> He would like to handle this data from receiver panel (not from admin)
>> He would like to have the "public profiles" on LeakDirectory to be two separate, one for IRPILeaks and one for ExpoLeaks

Thank you, I feel the design proposed fit both the needs. Admin require to keep the possibility, because in certain cases that I don't want mention here, Admin do more than Receivers. IRPI/EXPO, they are two different nodes. Right ? 

>> I didn't asked for "which information" do they expect to insert

Good, but, this feature is primarily in our interest.

>> The very same inquiry would be useful to be done/collected from other adopters, to collect their hypotetical practical use-case scenario.

I agree, I'll prepare a inquiry, but that can just influence partially this design. this is something we need, despite what they feel.

>> i've never requested a specification of the micro feature

but doing has been helpful. 

>> this is due to the fact that on each deliverable we have an amount of time allocated (e.g. D7.3 see only 40 men days allocates) so that it's important to have a roadmp and a clear idea of the things that we want to see addressed at the end of this limited time.

do not mistake the D7.3 with this feature. this is just a little part (two days on 40), which outcome will be likely part of D7.3, and will accomplish a couple of other needs (our historical overview of the globaleaks network, Hivos deliverable on success cases). I agree D7.3 require a bigger specification, but I was just making estimation of the couple of things I can address now.

>> but ok if you want start from D7 that we have currenly not defined this is the things that i would like to see defined before start.
>> https://docs.google.com/a/apps.globaleaks.org/document/d/1eCDxHfNrxe5Wwvy9Le-qd6_WLEThrWqANfPhzhAy-dM/edit

I can do it in the next weeks or later.

**Comment by vodkina** [comment]
I agree with those saying that this feature is mainly uneffective just now. 
Nevertheless, the discussion on possible implementations is a good point.
I postpone this to March Milestone, but I'd largely prefer the attention would be better focused on more urgent features and tasks.

**Comment by vecna** [comment]
Specification under review https://docs.google.com/document/d/1RGsqQDJ03f3iU4onnG7tBqZzTWNqBPRNF02XgruazHg/edit
